File: ums.client/src/shared/models/Subject.ts
export class Subject {
  id: number;
  name: string | null;
  credit: number;
  teacherId?: number;
  creatorUserId: string;
  creationDate?: Date | null;
  modificationUserId?: string | null;
  modificationDate?: Date | null;
  isDeleted: boolean;
  deletionUserId?: string | null;
  deletionDate?: Date | null;

  constructor(
    id: number,
    name: string | null,
    credit: number,
    isDeleted: boolean,
    creatorUserId: string,
    teacherId?: number,
    creationDate?: Date | null,
    modificationUserId?: string | null,
    modificationDate?: Date | null,
    deletionUserId?: string | null,
    deletionDate?: Date | null
  ) {
    this.id = id;
    this.name = name;
    this.credit = credit;
    this.teacherId = teacherId;
    this.creatorUserId = creatorUserId;
    this.creationDate = creationDate;
    this.modificationUserId = modificationUserId;
    this.modificationDate = modificationDate;
    this.isDeleted = isDeleted;
    this.deletionUserId = deletionUserId;
    this.deletionDate = deletionDate;
  }
}

File: UMS.Core/Models/Subject.cs
using System;
using System.Collections.Generic;

namespace UMS.Core.Models;

public partial class Subject
{
    public int Id { get; set; }

    public string Name { get; set; } = null!;

    public int Credit { get; set; }

    public int? TeacherId { get; set; }

    public string CreatorUserId { get; set; } = null!;

    public DateTime? CreationDate { get; set; }

    public string? ModificationUserId { get; set; }

    public DateTime? ModificationDate { get; set; }

    public bool IsDeleted { get; set; }

    public string? DeletionUserId { get; set; }

    public DateTime? DeletionDate { get; set; }

    public virtual ICollection<StudentSubject> StudentSubjects { get; set; } = new List<StudentSubject>();

    public virtual Teacher? Teacher { get; set; }
}
